wheel/generate_type_stubs.py

```diff
@@ -55,19 +55,19 @@ def __repr__(self) -> str: ...
     def __richcmp__(self) -> Any: ...
     def __deepcopy__(self) -> {name}: ...
     def __copy__(self) -> {name}: ...
-    @staticmethod
-    def from_bytes(bytes) -> {name}: ...
-    @staticmethod
-    def from_bytes_unchecked(bytes) -> {name}: ...
-    @staticmethod
-    def parse_rust(ReadableBuffer, bool = False) -> Tuple[{name}, int]: ...
+    @classmethod
+    def from_bytes(cls, bytes) -> {name}: ...
```

Comment:
Does `cls` not need a type? something like `Type[X]` where `X = TypeVar("X", bound="{name}")`.
I would also expect the return value to be `X`
In practice, `X` would need a unique name, probably derived from `{name}` to avoid collisions.

Reply:
I don't believe so.
`cls` is a special keyword for class methods that is automatically passed in when it's called, like `self`.

Maybe I'm misunderstanding what you mean though?

Reply:
take this function as an example. the return value is not (necessarily) `{name}`. It's an object of the type `cls`

Reply:
and we know `cls` is a type that is `{name}` or something that derives from it, so we can define it as `bound`

Reply:
I still think this is relevant. Some of these functions *do* return the type `{name}` I suppose, but some of them don't. The type signature is wrong for the ones supporting `cls` (afaict).

Reply:
Updated return val to `typing.Self`